coderd/agentapi/boundary_logs.go

```diff
@@ -2,14 +2,56 @@ package agentapi
 
 import (
 	"context"
+	"time"
 
-	"golang.org/x/xerrors"
+	"github.com/google/uuid"
+
+	"cdr.dev/slog"
 
 	agentproto "github.com/coder/coder/v2/agent/proto"
 )
 
-type BoundaryLogsAPI struct{}
+type BoundaryLogsAPI struct {
+	Log         slog.Logger
+	WorkspaceID uuid.UUID
+}
+
+func (a *BoundaryLogsAPI) ReportBoundaryLogs(ctx context.Context, req *agentproto.ReportBoundaryLogsRequest) (*agentproto.ReportBoundaryLogsResponse, error) {
+	for _, l := range req.Logs {
+		var logTime time.Time
+		if l.Time != nil {
+			logTime = l.Time.AsTime()
+		}
+
+		switch r := l.Resource.(type) {
+		case *agentproto.BoundaryLog_HttpRequest_:
+			if r.HttpRequest == nil {
+				continue
+			}
+
+			fields := []slog.Field{
+				slog.F("decision", allowBoolToString(l.Allowed)),
+				slog.F("workspace_id", a.WorkspaceID.String()),
+				slog.F("http_method", r.HttpRequest.Method),
+				slog.F("http_url", r.HttpRequest.Url),
+				slog.F("event_time", logTime.Format(time.RFC3339Nano)),
```

Comment:
How are the other fields injected?

- owner
- workspace_name
- agent_name
- request_id

Does the logger already have these?